test/apps/iodemo/io_demo.cc

```diff
@@ -897,8 +902,16 @@ class DemoClient : public P2pDemoCommon {
 
         if (msg->op >= IO_COMP_MIN) {
             assert(msg->op == IO_WRITE_COMP);
-            ++_num_completed;
-            ++_server_info[get_server_index(conn)].num_completed[IO_WRITE];
+
+            size_t server_index = get_server_index(conn);
```

Comment:
maybe should do _server_index_lookup.erase(conn); after "delete conn", in terminate_connection
to make sure all is flushed

Reply:
I would avoid to use conn after delete even as a key in the map.
I dont think it's possible but safer to check return value - it should not affect perf.